doc/api/process.md

```diff
@@ -1684,6 +1684,107 @@ the [`'warning'` event][process_warning] and the
 [`emitWarning()` method][process_emit_warning] for more information about this
 flag's behavior.
 
+## `process.parseArgs([argv[, options]])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `argv` {Array<string>|Object} (Optional) Array of argument strings; defaults
+  to [`process.argv.slice(2)`](process_argv). If an Object, the default is used,
+  and this parameter is considered to be the `options` parameter.
+* `options` {Object} (Optional) The `options` parameter, if present, is an
+  object supporting the following property:
+  * `expectsValue` {Array<string>|string} (Optional) One or more argument
+    strings which _expect a value_ when present
+* Returns: {Object} An object having properties corresponding to parsed Options
+  and Flags, and a property `_` containing Positionals
+
+The `process.parseArgs` function parses command-line arguments from an array of
+strings and returns an object representation.
+
+Example using [`process.argv`][]:
+
+```js
+// script.js
+// called via `node script.js --foo bar baz`
+const argv = process.parseArgs();
+
+// argv.foo === true
+if (argv.foo) {
+  console.log(argv._); // prints [ 'bar', 'baz' ]
+}
+```
+
+Example using a custom `argv` and the `expectsValue` option:
+
+```js
+const argv = process.parseArgs(
+  ['--foo', 'bar', 'baz'],
+  { expectsValue: ['foo'] }
+);
+
+// argv.foo === 'bar'
+if (argv.foo === 'bar') {
+  console.log(argv._); // prints [ 'baz' ]
+}
+```
+
+[`ERR_INVALID_ARG_TYPE`][] will be thrown if the `argv` parameter is not an
+Array.
+
+Arguments fall into one of three catgories:
+
+* _Flags_, which begin with one or more dashes (`-`), and _do not_ have an
+  associated string value (e.g., `node app.js --verbose`)
+  * These will be parsed automatically; you do not need to "declare" them
+  * The Flag _name_ is the string following the prefix of one-or-more dashes,
+    e.g., the name of `--foo` is `foo`
+  * Flag names become property names in the returned object
+  * When appearing _once_ in the array, the value of the property will be `true`
+  * When _repeated_ in the array, the value of the property becomes a count of
+    repetitions (e.g., `['-v', '-v' '-v']` results in `{ v: 3 }`)
+* _Options_, declared by `expectsValue`, which begin with one or more dashes,
+  and _do_ have an associated value (e.g., `node app.js --require script.js`)
+  * Use the `expectsValue` option to `process.parseArgs` to declare Options
+  * The Option _name_ is the string following the prefix of one-or-more dashes,
+    e.g., the name of `--foo` is `foo`
+  * The Option _value_ is the next string following the name, e.g., the Option
+    value of `['--foo' 'bar']` is `bar`
+  * Option values may be provided _with or without_ a `=` separator (e.g.,
+    `['--require=script.js']` is equivalent to `['--require', 'script.js']`)
```

Comment:
What happens if the user don't pass `expectsValue` and passes `['--require=scripts.js']` to processArgs?

Reply:
it becomes `true`; updated docs to reflect this.